src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/IntervalTreeHelpers.cs

```diff
@@ -31,49 +33,14 @@ internal interface IIntervalTreeWitness<T, TIntervalTree, TNode>
 /// <summary>
 /// Utility helpers used to allow code sharing for the different implementations of <see cref="IIntervalTree{T}"/>s.
 /// </summary>
-internal static class IntervalTreeHelpers<T, TIntervalTree, TNode, TIntervalTreeWitness>
+internal static partial class IntervalTreeHelpers<T, TIntervalTree, TNode, TIntervalTreeWitness>
     where TIntervalTree : IIntervalTree<T>
     where TIntervalTreeWitness : struct, IIntervalTreeWitness<T, TIntervalTree, TNode>
 {
     private static readonly ObjectPool<Stack<TNode>> s_nodeStackPool = new(() => new(), 128, trimOnFree: false);
 
-    public static IEnumerator<T> GetEnumerator(TIntervalTree tree)
```

Comment:
so it is nice that this becomes a struct-enumerator and has no more code for itself anymore.